docs/source/tutorials_source/package/tutorial_custom_augmentations.py

```diff
@@ -0,0 +1,366 @@
+
+        # use linear interpolation of cdf to find new pixel values
+        image_equalized = np.interp(image.flatten(), bins[:-1], cdf)
+        return Image.fromarray(image_equalized.reshape(image.shape))
+
+# %%
+# Since we can't use color jitter on X-ray images, let's replace it and add some
+# Gaussian noise instead. It's easiest to apply this after the image has been
+# converted to a PyTorch tensor.
+
+class GaussianNoise:
+    """Applies random Gaussian noise to a tensor.
+
+    The intensity of the noise is dependent on the mean of the pixel values.
+    See https://arxiv.org/pdf/2101.04909.pdf for more information.
+
+    """
+
+    def __call__(self, sample: torch.Tensor) -> torch.Tensor:
+        mu = sample.mean()
+        snr = np.random.randint(low=4, high=8)
+        sigma = mu / snr
+        noise = torch.normal(torch.zeros(sample.shape), sigma)
+        return sample + noise
+
+
+# %%
+# Now that we have implemented our custom augmentations, we can combine them
+# with available augmentations from the torchvision library. Make sure, that
+# the first augmentation is the histogram normalization, and that the Gaussian
+# noise is applied after converting the image to a tensor.
+#
+# Note that we also transform the image from grayscale to RGB by simply repeating
+# the single color channel three times. The reason for this is that our ResNet expects
+# a three color channel input. This step can be skipped if a different backbone network
+# is used.
+
+
+# compose the custom augmentations with available augmentations
+transform = torchvision.transforms.Compose([
+    HistogramNormalize(),
+    torchvision.transforms.Grayscale(num_output_channels=3),
+    torchvision.transforms.RandomResizedCrop(size=input_size, scale=(0.2, 1.0)),
+    torchvision.transforms.RandomHorizontalFlip(p=0.5),
+    torchvision.transforms.RandomVerticalFlip(p=0.5),
+    torchvision.transforms.GaussianBlur(21),
+    torchvision.transforms.ToTensor(),
+    GaussianNoise(),
+])
+
+
+# %%
+# Let's take a look at what our augmentation pipeline does to an image!
+# We plot the original image on the left and two random augmentations on the 
+# right.
+
+example_image_name = '55e8e3db7309febee415515d06418171.tiff'
+example_image_path = os.path.join(path_to_data, example_image_name)
+example_image = np.array(Image.open(example_image_path))
+
+# torch transform returns a 3 x W x H image, we only show one color channel
+augmented_image_1 = transform(example_image).numpy()[0]
+augmented_image_2 = transform(example_image).numpy()[0]
+
+fig, axs = plt.subplots(1, 3)
+
+axs[0].imshow(example_image)
+axs[0].set_axis_off()
+axs[0].set_title('Original Image')
+
+axs[1].imshow(augmented_image_1)
+axs[1].set_axis_off()
+
+axs[2].imshow(augmented_image_2)
+axs[2].set_axis_off()
+
+# %%
+# Finally, in order to use the augmentation pipeline we defined for self-supervised
+# learning, we need to create a Lightly collate function like so:
```

Comment:
Let's use `lightly` for anything related to the pip package and `Lightly` if we talk about the company :)

Reply:
ahh I always forget...